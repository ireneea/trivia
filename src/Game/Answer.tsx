import React from "react";
import { View, Text, StyleSheet, StyleProp, TouchableWithoutFeedback } from "react-native";

import { AnswerType } from "../ts/appTypes";
import { fontSize, colors } from "../styles/globals";

type Props = {
  answer: AnswerType;
  onSelect(): void;
  isCorrect?: boolean;
  isIncorrect?: boolean;
  feedback?: boolean;
  selected?: boolean;
};

const Answer: React.FC<Props> = (props) => {
  const { answer, onSelect, feedback } = props;

  const appliedStyle = getContainerStyle(props);

  return (
    <TouchableWithoutFeedback onPress={onSelect} disabled={feedback}>
      <View style={appliedStyle.container} testID={`answer-${answer.answer}`}>
        <Text style={appliedStyle.text} accessibilityLabel={answer.answer}>
          {answer.answer}
        </Text>
      </View>
    </TouchableWithoutFeedback>
  );
};

const getContainerStyle = (props: Props): any => {
  const { isCorrect = false, isIncorrect = false, feedback = false, selected = false } = props;

  let container: StyleProp<{}> | [StyleProp<{}>];
  let text: StyleProp<{}> | [StyleProp<{}>];

  if (feedback && isCorrect) {
    container = correctAnswerContainer;
    text = correctAnswerText;
  } else if (selected && feedback && isIncorrect) {
    container = incorrectAnswerContainer;
    text = incorrectAnswerText;
  } else {
    container = styles.answerContainer;
    text = styles.answerText;
  }

  return { container, text };
};

const styles = StyleSheet.create({
  answerContainer: {
    backgroundColor: colors.secondary,
    marginVertical: 10,
    paddingVertical: 15,
    paddingHorizontal: 15,
    borderRadius: 8,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,

    elevation: 3,
  },
  answerText: {
    color: colors.normal,
    ...fontSize.l,
  },
  correctAnswerContainer: {
    backgroundColor: colors.green,
  },
  incorrectAnswerContainer: {
    backgroundColor: colors.red,
  },
  correctAnswerText: {
    // color: colors.normal,
  },
  incorrectAnswerText: {
    // color: colors.normal,
  },
});

export const correctAnswerContainer = StyleSheet.compose(styles.answerContainer, styles.correctAnswerContainer);

export const incorrectAnswerContainer = StyleSheet.compose(styles.answerContainer, styles.incorrectAnswerContainer);

const correctAnswerText = StyleSheet.flatten([styles.answerText, styles.correctAnswerText]);

const incorrectAnswerText = StyleSheet.flatten([styles.answerText, styles.incorrectAnswerText]);

export default Answer;
