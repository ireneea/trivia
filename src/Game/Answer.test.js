import React from "react";
import { render } from "@testing-library/react-native";

import Answer from "./Answer";

describe("Answer", () => {
  const answer = { answer: "one" };
  it("render Answer correctly", () => {
    const { baseElement } = render(<Answer answer={answer} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("displays the answer", () => {
    const { getByText } = render(
      <Answer answer={{ ...answer, answer: "two" }} />
    );
    expect(getByText("two")).toBeTruthy();
  });
});
