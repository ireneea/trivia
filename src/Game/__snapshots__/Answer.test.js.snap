// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer render Answer correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "deepskyblue",
          "borderRadius": 5,
          "marginVertical": 10,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "snow",
          }
        }
      >
        one
      </Text>
    </View>
  </View>
</View>
`;
